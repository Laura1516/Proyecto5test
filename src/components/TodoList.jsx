import React from 'react';
import TodoItem from './TodoItem';

function TodoList({ tasks, onToggle }) {
    return (
        <ul>
            {tasks.map((task, index) => (
                <TodoItem
                    key={index}
                    task={task.text}         // Muestra el texto
                    completed={task.completed} // Indica si estÃ¡ completado
                    onToggle={() => onToggle(index)} // Maneja el click
                />
            ))}
        </ul>
    );
}

export default TodoList;