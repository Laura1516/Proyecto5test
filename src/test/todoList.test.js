import React from 'react';
import { render } from '@testing-library/react';
import TodoList from '../components/TodoList';


test('Genera un snapshot del componente TodoList', () => {
    const tasks = ['Tarea 1', 'Tarea 2'];
    const { asFragment } = render(<TodoList tasks={tasks} />);
    expect(asFragment()).toMatchSnapshot();
});