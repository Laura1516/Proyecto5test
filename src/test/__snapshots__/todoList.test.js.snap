// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Genera un snapshot del componente TodoList 1`] = `
<DocumentFragment>
  <ul>
    <li
      class=""
      style="cursor: pointer;"
    />
    <li
      class=""
      style="cursor: pointer;"
    />
  </ul>
</DocumentFragment>
`;
